moto/route53/responses.py

```diff
@@ -53,15 +53,16 @@ def list_hosted_zones_by_name_response(self, request, full_url, headers):
         dnsname = query_params.get("dnsname")
 
         if dnsname:
-            dnsname = dnsname[
-                0
-            ]  # parse_qs gives us a list, but this parameter doesn't repeat
-            # return all zones with that name (there can be more than one)
+            dnsname = dnsname[0]
+            if dnsname[-1] != ".":
+                dnsname += "."
```

Comment:
Might be me misreading the test, but I can't actually see a test case where we hit this if-statement?

Reply:
Added add'l test in 2abf91cab05ef7a903fb393018dd8bd0c6c656d4